sdk/communication/azure-communication-callingserver/src/main/java/com/azure/communication/callingserver/ServerCallAsync.java

```diff
@@ -0,0 +1,511 @@
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
+            return this.serverCallInternal.recordingStateAsync(serverCallId, recordingId)
+                .onErrorMap(CommunicationErrorException.class, e -> CallingServerErrorConverter.translateException(e));
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Get recording state
+     *
+     * @param recordingId The recording id to stop.
+     * @return response for a successful getRecordingState request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<CallRecordingStateResponse>> getRecordingStateWithResponse(String recordingId) {
+        return getRecordingStateWithResponse(recordingId, Context.NONE);
+    }
+
+    Mono<Response<CallRecordingStateResponse>> getRecordingStateWithResponse(String recordingId, Context context) {
+        try {
+            Objects.requireNonNull(recordingId, "'recordingId' cannot be null.");
+            return withContext(contextValue -> {
+                contextValue = context == null ? contextValue : context;
+                return this.serverCallInternal
+                    .recordingStateWithResponseAsync(serverCallId, recordingId, contextValue)
+                    .onErrorMap(CommunicationErrorException.class, e -> CallingServerErrorConverter.translateException(e));
+            });
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Play audio in a call.
+     *
+     * @param audioFileUri The media resource uri of the play audio request. Currently only Wave file (.wav) format
+     *                     audio prompts are supported. More specifically, the audio content in the wave file must
+     *                     be mono (single-channel), 16-bit samples with a 16,000 (16KHz) sampling rate.
+     * @param audioFileId Tne id for the media in the AudioFileUri, using which we cache the media resource.
+     * @param callbackUri The callback Uri to receive PlayAudio status notifications.
+     * @param operationContext The operation context.
+     * @return the response payload for play audio operation.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PlayAudioResponse> playAudio(String audioFileUri,
+                                             String audioFileId,
+                                             String callbackUri,
+                                             String operationContext) {
+        try {
+            Objects.requireNonNull(audioFileUri, "'audioFileUri' cannot be null.");
+
+            //Currently we do not support loop on the audio media for out-call, thus setting the loop to false
+            PlayAudioRequest playAudioRequest = new PlayAudioRequest();
+            playAudioRequest.setAudioFileUri(audioFileUri);
```

Comment:
Use fluent style coding, don't need to repeat playAudioRequest

Reply:
using fluent